### ISSUE TITLE
"Invalid UTF-8 Stream" errors in fast-tags 2.0.1

### ISSUE THREAD

**Issue by mpickering** [elaforge/fast-tags#62]
I now get quite a few errors when trying to use fast-tags on GHC's code base which I didn't in older versions of fast-tags. Seems something to do with encoding, help?

```
[matt@nixos:~/ghc-vanessa]$ fast-tags -Rv compiler/
3: compiler/GHC/Builtin/PrimOps.hs                                            Error while analyzing compiler/GHC.hs:
Cannot decode byte '\x88': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
10: compiler/GHC/ByteCode/Asm.hs                                              or while analyzing compiler/GHC/ByteCode/Instr.hs:
Cannot decode byte '\x90': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
17: compiler/GHC/Cmm/CLabel.hs                                                nalyzing compiler/GHC/ByteCode/InfoTable.hs:
Cannot decode byte '\xc1': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
30: compiler/GHC/Cmm/Graph.hs                                                 Error while analyzing compiler/GHC/Cmm/Info.hs:
Cannot decode byte '\x90': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream
Error while analyzing compiler/GHC/Cmm/Info/Build.hs:
Cannot decode byte '\xd0': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream
35: compiler/GHC/Cmm/LayoutStack.hs                                           Error while analyzing compiler/GHC/Cmm/Lexer.x:
Cannot decode byte '\x90': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
51: compiler/GHC/Cmm/Switch.hs                                                ompiler/GHC/CmmToAsm/Dwarf/Types.hs:
Cannot decode byte '\xe9': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
52: compiler/GHC/Cmm/Switch/Implement.hs                                      Error while analyzing compiler/GHC/CmmToAsm/BlockLayout.hs:
Cannot decode byte '\xfc': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
Error while analyzing compiler/GHC/CmmToAsm/PPC/Instr.hs:
Cannot decode byte '\xe8': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
58: compiler/GHC/CmmToAsm/AArch64/Cond.hs                                     Error while analyzing compiler/GHC/CmmToAsm/Reg/Linear.hs:
Cannot decode byte '\xf8': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream
64: compiler/GHC/CmmToAsm/CFG.hs                                              Error while analyzing compiler/GHC/CmmToAsm/Reg/Liveness.hs:
Cannot decode byte '\xd0': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream
70: compiler/GHC/CmmToAsm/Dwarf/Constants.hs                                  nalyzing compiler/GHC/CmmToAsm/Utils.hs:
73: compiler/GHC/CmmToAsm/Instr.hs                                            eUtf8: Invalid UTF-8 stream
77: compiler/GHC/CmmToAsm/PPC/CodeGen.hs                                       while analyzing compiler/GH78: compiler/GHC/CmmToAsm/PPC/Cond.hs                                         se.hs:
Cannot decode byte '\xbc': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
111: compiler/GHC/CmmToAsm/X86/Cond.hs                                        s:pt/SpecConstr.h
Cannot decode byte '\xce': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
Error while analyzing compiler/GHC/Core/Opt/Specialise.hs:
Cannot decode byte '\xe0': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
Error while analyzing compiler/GHC/Core/TyCo/FVs.hs:
114: compiler/GHC/CmmToAsm/X86/RegInfo.hs                                      UTF-8 stream
116: compiler/GHC/CmmToC.hs                                                   r while analyzing compiler/GHC/Core/Type.hs:
117: compiler/GHC/CmmToLlvm.hs                                                alid UTF-8 stream
119: compiler/GHC/CmmToLlvm/CodeGen.hs                                        ile analyzing compiler/GHC/C124: compiler/GHC/CmmToLlvm/Regs.hs                                           :oIface.hs
128: compiler/GHC/Core/Coercion/Axiom.hs                                      ecode byte '\xb8': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
Error while analyzing compiler/GHC/Data/Graph/Ops.hs:
Cannot decode byte '\xd0': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream
159: compiler/GHC/Core/Opt/Simplify/Utils.hs                                  Error while analyzing compil163: compiler/GHC/Core/Opt/WorkWrap.hs                                        river/Flags.hs:
Cannot decode byte '\xf0': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
165: compiler/GHC/Core/PatSyn.hs                                              Error while analyzing compiler/GHC/Driver/Make.hs:
Cannot decode byte '\x90': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
Error while analyzing compiler/GHC/Driver/Errors/Types.hs:
Cannot decode byte '\x81': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
172: compiler/GHC/Core/SimpleOpt.hs                                           Error while analyzing compiler/GHC/HsToCore/Arrows.hs:
Cannot decode byte '\x90': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream
Lexical error while in state AlexCode {unAlexCode = 0} at line 684: Iv'
175: compiler/GHC/Core/Tidy.hs                                                Lexical error while in state AlexCode {unAlexCode = 0} at line 389: ''
180: compiler/GHC/Core/TyCo/Tidy.hs                                           Error while analyzing compiler/GHC/HsToCore/Pmc/Solver/Types.hs:
Cannot decode byte '\xa2': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
182: compiler/GHC/Core/TyCon/Env.hs                                           Error while analyzing compiler/GHC/Iface/Ext/Utils.hs:
Cannot decode byte '\x90': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream
184: compiler/GHC/Core/TyCon/Set.hs                                           le analyzing compiler/GHC/Iface/Make.hs:
Cannot decode byte '\xe0': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
Error while analyzing compiler/GHC/Iface/Recomp.hs:
Cannot decode byte '\x98': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream
190: compiler/GHC/Core/Utils.hs                                               Error while analyzing compiler/GHC/Parser/Annotation.hs:
Cannot decode byte '\x90': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
192: compiler/GHC/CoreToStg.hs                                                Error while analyzing compil194: compiler/GHC/Data/Bag.hs                                                 /PostProcess.hs:
Cannot decode byte '\x83': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
198: compiler/GHC/Data/EnumSet.hs                                             Error while analyzing compiler/GHC/Platform/RISCV64.hs:
Cannot decode byte '\x88': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
204: compiler/GHC/Data/Graph/Color.hs                                         Error while analyzing compiler/GHC/Rename/Pat.hs:
205: compiler/GHC/Data/Graph/Directed.hs                                      t decode byte '\xa2': Data.T211: compiler/GHC/Data/Maybe.hs                                               amUtf8: Invalid UTF-8 stream
212: compiler/GHC/Data/OrdList.hs                                             Error while analyzing compiler/GHC/Rename/Utils.hs:
Cannot decode byte '\xe0': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
219: compiler/GHC/Data/UnionFind.hs                                           Error while analyzing compil242: compiler/GHC/Driver/Config/Stg/Pipeline.hs                               :ig.hsrTags/Tag
247: compiler/GHC/Driver/Env/KnotVars.hs                                      ernal.Encoding.decodeUtf8: I250: compiler/GHC/Driver/Errors/Ppr.hs                                        alid UTF-8 stream
275: compiler/GHC/Hs/Expr.hs                                                  asks.hs:ysTools/Tz
315: compiler/GHC/HsToCore/Utils.hs                                           eamUTF-8 s Inval
346: compiler/GHC/Llvm.hs                                                     C/Tc/Gen/Bind.hs:
359: compiler/GHC/Parser/Lexer.x                                              .Fusion.streamUtf8: Invalid UTF-8 stream
361: compiler/GHC/Parser/PostProcess/Haddock.hs                               /Tc/Gen/Foreign.hs:
374: compiler/GHC/Platform/S390X.hs                                           TF-8 streamf8: Invalid U
409: compiler/GHC/Stg/InferTags/Rewrite.hs                                    h.hs:Tile
Cannot decode byte '\x90': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
443: compiler/GHC/StgToCmm/TagCheck.hs                                        nstance.hs:/TyC
452: compiler/GHC/SysTools/Process.hs                                         annot decode byte '\xaa': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream
Error while analyzing compiler/GHC/Tc/Types/Constraint.hs:
475: compiler/GHC/Tc/Gen/HsType.hs                                            ode byte '\x9a': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
503: compiler/GHC/Tc/Types/Evidence.hs                                        :hss/c/U
507: compiler/GHC/Tc/Utils/Concrete.hs                                         '\x9b': Data.Text.Internal.511: compiler/GHC/Tc/Utils/TcMType.hs                                         sion.streamUtf8: Invalid UTF-8 stream
536: compiler/GHC/Types/Id.hs                                                 ler/GHC/Types/IPE.hs:
Cannot decode byte '\x90': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
Error while analyzing compiler/GHC/Types/Id/Info.hs:
Cannot decode byte '\xd3': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
540: compiler/GHC/Types/Meta.hs                                               Error while analyzing compiler/GHC/Types/Name.hs:
Cannot decode byte '\x90': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
549: compiler/GHC/Types/PkgQual.hs                                            Error while analyzing compiler/GHC/Types/RepType.hs:
Cannot decode byte '\x83': Data.Text.Internal.Encoding.Fusion.streamUtf8: Invalid UTF-8 stream
556: compiler/GHC/Types/Target.hs                                             Error while analyzing compiler/GHC/Types/SourceFile.hs:
Cannot decode byte '\xd5': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream
606: compiler/GHC/Utils/FV.hs                                                  0} at line 0: ''
612: compiler/GHC/Utils/Logger.hs                                             Error while analyzing compiler/GHC/Utils/Misc.hs:
Cannot decode byte '\x90': Data.Text.Internal.Encoding.decodeUtf8: Invalid UTF-8 stream
632: compiler/Setup.hs                                                       
```

**Comment by sergv** [comment]
* Is your `fast-tags` built from master or from hackage?
* Which ghc verson did you build with?
* Could you please specify versions of other versions dependencies used to build the fast-tags exe?

**Comment by sergv** [comment]
The problem is that latest released `fast-tags-2.0.1` can only be built with GHC 9.0 which, without a recent fix, has the problem outlined above. There are fixes to make it build with 9.2 and fixes for working correctly with 9.0 but they're not released.

@elaforge could you please upload a new fast-tags release to Hackage and also edit metadata on `fast-tags-2.0.1` to prohibit building it with GHC 9.0 (thus forcing users to use 8.10.7 which is not as susceptible to the problem). Or just deprecate `fast-tags-2.0.1` since the next version should be preferred. Or add me to package maintainers on Hackage and I can do all of the above.

**Comment by sergv** [comment]
Fixed in #63 as part of #61.
